pkg/geo/geomfn/topology_operations.go

```diff
@@ -188,3 +193,75 @@ func SharedPaths(a geo.Geometry, b geo.Geometry) (geo.Geometry, error) {
 	}
 	return gm, nil
 }
+
+// Node Returns a geometry containing a set of linestrings using the least possible number of nodes while preserving all of the input ones.
+func Node(g geo.Geometry) (geo.Geometry, error) {
+	if g.ShapeType() != geopb.ShapeType_LineString && g.ShapeType() != geopb.ShapeType_MultiLineString {
+		return geo.Geometry{}, errors.New("geometry type is unsupported. Please pass a LineString or a MultiLineString")
+	}
+	ep, err := extractUniqueEndpoints(g)
```

Comment:
nit: move this down closer to where it's actually used

Reply:
done